### PR TITLE
Implement input component.

### PR DESCRIPTION
**Pull request by leyanessantiago**

### CODE REVIEW

File: src/components/Input/error.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export default function Help({ error }) {
```
Comment (LeandroTorresSicilia) [inline src/components/Input/error.js]: Error

File: src/components/Input/icon.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+export default function Icon(props) {
+    const {
+        isVisible,
+        iconName,
+        position,
+    } = props;
+    if (isVisible && iconName) {
+        if (position === 'left') {
+            return (
+                <IconSvg iconName={iconName} className="slds-icon slds-input__icon slds-input__icon_left slds-icon-text-default" />
```
Comment (LeandroTorresSicilia) [inline src/components/Input/icon.js]: you can return only one and call getIconClassName function

File: src/components/Input/index.js [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+        if (error) {
+            return 'slds-has-error';
+        }
+        return null;
+    };
+
+
+    const getLabelClassNames = () => classnames('slds-form-element__label', labelClassName);
+
+    const getContainerClassNames = () => classnames('slds-form-element', getErrorContainerClassNames(), className);
```
Comment (LeandroTorresSicilia) [inline src/components/Input/index.js]: you can write this function like this:
```
const getContainerClassNames = () => classnames('slds-form-element', {
    'slds-has-error': error,
}, className);
```

File: src/components/Input/index.js [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+                    className={getInputClassNames()}
+                    value={value}
+                    placeholder={placeholder}
+                    onChange={onChange}
+                    tabIndex={tabIndex}
+                    disabled={disabled}
+                    readOnly={readonly}
+                    required={required}
+                    maxLength={maxLength}
+                    minLength={minlength}
```
Comment (LeandroTorresSicilia) [inline src/components/Input/index.js]: Here the prop should be minLength instead of minlength. L uppercase

File: src/components/Input/index.js [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+        </div>
+    );
+}
+
+Input.propTypes = {
+    /** The class name of the root element. */
+    className: PropTypes.string,
+    /** It is an object with custom style applied to the root element. */
+    style: PropTypes.object,
+    value: PropTypes.string,
```
Comment (LeandroTorresSicilia) [inline src/components/Input/index.js]: add description for all props

File: src/components/Input/index.js [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+    labelClassName: PropTypes.string,
+    /** The name of the icon. Names are written in the
+     format '\utility:down\' where 'utility' is the category, and 'down' is the
+     specific icon to be displayed. */
+    iconName: PropTypes.string,
+    /** The position of the icon if it is passed. */
+    iconPosition: PropTypes.oneOf([
+        'left', 'right',
+    ]),
+    help: PropTypes.node,
```
Comment (reiniergs) [inline src/components/Input/index.js]: change this props name to `bottomHelpText`

File: src/components/Input/index.js [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+    type: 'text',
+    maxLength: undefined,
+    minlength: undefined,
+    pattern: undefined,
+    labelClassName: undefined,
+    iconName: '',
+    iconPosition: 'left',
+    help: null,
+    bare: false,
+    centered: false,
```
Comment (reiniergs) [inline src/components/Input/index.js]: this props (bare, centered) are not present in the HTML input, we can use `isBare`, `isCentered`

File: src/components/Input/index.js [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+        return null;
+    };
+
+    const getFormControlClassNames = () => classnames('slds-form-element__control', getIconContainerClassNames());
+
+    const isLeftIcon = iconPosition === 'left';
+    const isRightIcon = iconPosition === 'right';
+    return (
+        <div className={getContainerClassNames()} style={style}>
+            <label className={getLabelClassNames()}>{label}</label>
```
Comment (reiniergs) [inline src/components/Input/index.js]: are we implementing the required styles here??

**Pull request merged**